android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java

```diff
@@ -616,6 +621,59 @@ public void run() {
         }.start();
     }
 
+    /**
+     * Registers a wifi network in suggested networks of the application,
+     * so connections can then be made preferentially to this. For API >= 30
+     * also the intent to permanently store such network in user configuration
+     * can be executed.
+     * *** registerWifiNetwork :
+     * param ssid, SSID to register
+     * param password, passphrase to use
+     * param security, security mode (WPA or null) to use
+     * param permanent_add, (default {@code false}) call also intent to save to system (>=30 only)
+     * return {@code true} if the operation succeeds, {@code false} otherwise
+     */
+    private void registerWifiNetwork(final MethodCall poCall, final Result poResult) {
+        String ssid = poCall.argument("ssid");
+        String password = poCall.argument("password");
+        String security = poCall.argument("security");
+        Boolean permanentAdd = poCall.argument("permanent_add");
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            final WifiNetworkSuggestion.Builder suggestedNet = new WifiNetworkSuggestion.Builder();
+            suggestedNet.setSsid(ssid);
+
+            if (security != null && security.toUpperCase().equals("WPA")) {
+                suggestedNet.setWpa2Passphrase(password);
+            } else if (security != null && security.toUpperCase().equals("WEP")) {
+                // WEP is not supported
+                poResult.error("Error", "WEP is not supported for Android SDK " + Build.VERSION.SDK_INT, "");
+            }
+
+            final ArrayList<WifiNetworkSuggestion> suggestionsList = new ArrayList<WifiNetworkSuggestion>();
+            suggestionsList.add(suggestedNet.build());
+
+            moWiFi.addNetworkSuggestions(suggestionsList);
+
+            if (moWiFi.addNetworkSuggestions(suggestionsList) != WifiManager.STATUS_NETWORK_SUGGESTIONS_SUCCESS) {
+                poResult.error("Error", "Network suggestion not accepted", "");
+            } else {
+                if (permanentAdd && (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R)) { // Intent just available on >= 30
+                    Bundle bundle = new Bundle();
+                    bundle.putParcelableArrayList(android.provider.Settings.EXTRA_WIFI_NETWORK_LIST, suggestionsList);
+                    Intent intent = new Intent(android.provider.Settings.ACTION_WIFI_ADD_NETWORKS);
+                    intent.putExtras(bundle);
+                    intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+                    moContext.startActivity(intent);
+                }
+            }
+
+            poResult.success(null);
+        } else {
+            poResult.error("Error", "WifiNetworkSuggestion is not supported for Android SDK " + Build.VERSION.SDK_INT, "");
+        }
+    }
+
```

Comment:
This code in `connectToDeprecated` is the actual code for "registering network" (it saves the network in android system, also making a connection to it).


The following code in the method does this:
```java
 int updateNetwork = -1;

        if (mWifiConfigList != null) {
            for (android.net.wifi.WifiConfiguration wifiConfig : mWifiConfigList) {
                if (wifiConfig.SSID.equals(conf.SSID)) {
                    conf.networkId = wifiConfig.networkId;
                    updateNetwork = moWiFi.updateNetwork(conf);
                }
            }
        }

        /// If network not already in configured networks add new network
        if (updateNetwork == -1) {
            updateNetwork = moWiFi.addNetwork(conf);
            moWiFi.saveConfiguration();
        }

        if (updateNetwork == -1) {
            return false;
        }
```

We should handle the <Q case for this method by calling it, without checking if the network is connected or not (what is done in `connectToDeprecated`).

My suggestion is strip out this code (with proper return of result) to `registerWifiNetworkDeprecated` - call this in `connectToDeprecated` and `registerWifiNetwork`.